src/manifest.c

```diff
@@ -675,19 +675,21 @@ char *file_type_to_string(struct file *file)
 	return type;
 }
 
+/* Calculate the contentsize for the manifest based on file sizes.
+ * This should calculate the files uniquely included in this manifest, but none
+ * of its submanifests, which will allow calculation of all bundles on a system
+ * by adding all manifest->contentsizes of installed bundles.
+ */
```

Comment:
Thanks.

Just to clarify in more detail: What I described is the worse case scenario. In reality, bundles not in the same include chain (however you want to describe this) often just partially overlap, content-wise. At that point, swupd-client would need to know content size at the *file level* instead of *manifest level* to determine the real content size. Because manifests do not list content size at the file level, determining the real content size is intractable with the current state of things.